modules/cluster-logging-elasticsearch-retention.adoc

```diff
@@ -0,0 +1,104 @@
+
+////
+Only maxAge in current implementation:
+You can specify any of the following conditions:
+* `max_age`. The maximum age that is allowed before triggering a rollover for the _hot_ indices, or those ​indices that are actively being updated and queried. Specify a value in days (d), hours (h), minutes (m), or seconds (s). 
+* `max_docs`. The maximum number of documents allowed in an index before triggering a rollover. Documents added since the last refresh are not included in the document count. The document count does not include documents in replica shards.
+* `max_size`. The maximum size the index can be before a rollover is triggered. Specify a value in bytes (b), kilobytes (kb), megabytes (mb), or gigabytes (gb). 
+////
+
+For example, the following retention policy rolls over active indices for the infrastructure logs every three days and the old indices are deleted seven days after rollover. {product-title} checks every 15 minutes to determine if the infrastructure indices need to be rolled over.
+
+----
+apiVersion: "logging.openshift.io/v1"
+kind: "Elasticsearch"
+metadata:
+  name: "elasticsearch"
+spec:
+  indexManagement:
+    policies: <1>
+    - name: infra-policy
+      pollInterval: 15m
+      phases:
+        hot:
+          actions:
+            rollover:
+              maxAge:   3d
+        delete:
+          minAge: 7d
+    mappings: <2>
+    - name:  node.infra 
+      policyRef: infra-policy
+      aliases:
+      - infra  
+...
+----
+<1> For each log source, the retention policy indicated when to delete and rollover logs for that source. The policy includes: 
+* When to rollover the indices, determine by `maxAge`. This setting comes from the Cluster Logging CR.
+* When to check if rollover should be performed, determined by `pollInterval`. This setting is the default.
+* When the rolled-over indicies will be deleted after the rollover, determined by `minAge`. This setting is the default.
+<2> For each log source with a retention policy, the mapping connects the index, under alias, to the retention policy in `policyRef`.
+
+.Prerequisites
+
+* Cluster logging and Elasticsearch must be installed.
+
+.Procedure
+
+To configure the Elasticsearch retention policy:
+
+. Edit the Cluster Logging CR to add or modify the `retentionPolicy` parameter:
++
+----
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogging"
+...
+spec:
+  managementState: "Managed"
+  logStore:
+    type: "elasticsearch"
+    elasticsearch:
+      nodeCount: 2
+    retentionPolicy: <1>
+      application:
+        maxAge: 1d
+      infra:
+        maxAge: 7d
+      audit:
+        maxAge: 7d
+...
+----
+<1> Specify the maximum age that is allowed before triggering a rollover for each log source.
++
+The Elasticsearch Operator creates or modifies the `indexmanagement-scripts` configMap in the {PROJECT} namespace with the retention policy.
++
+----
+$ oc get cm
```

Comment:
@jcantrill Which project is the configMap in?

Reply:
The config-map `indexmanagement-scripts` is in `openshift-logging`.